0000-zmix/README.md

```diff
@@ -0,0 +1,200 @@
+For example, anonymous credentials uses the following public data: the schema that describes the credential attributes and an issuers related information like public cryptographic keys.
+Proof specs should be capable of being generated deterministically by any party.
+Used in combination with either a **witness** to generate a proof, or a **proof** to be validated.
+
+## Witness
+
+A document containing the Prover's private data. This includes: private keys, secrets, attribute values, hash pre-images.
+
+## Proof
+
+Cryptographic token(s) that is the proof to be verified.
+
+## Out of scope
+
+Initially, ZMix will not handle creating **proof specs**.
+In a ZKP protocol, a relying party will make a **proof request** for Provers to fulfill.
+A **proof request** contains non-cryptographic information like which credential issuers it will accept,
+how votes will be submitted, and subscriptions are checked. A Prover may not accept the **proof request** as is and could make
+a counter proposal in the form of a **proof offer**. Once a **proof request** is agreed upon, the Prover will have to choose
+how to fulfill the proof. Prover will have to acquire any missing information not already on hand. Prover may also choose between
+multiple credentials that can fulfill the proof. The choices were made will need to be communicated to the verifier as a **proof resolution**–the non-cryptographic
+description about the proof. The **proof resolution** and **proof request** can be combined with the public data to form the **proof spec**.
+Exactly where the public cryptographic material is stored and how it is retrieved is up to end consumers.
+This could be handled later.
+
+## Example diagram
+
+![flow](zmix_flow.png)
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+There are six ZKPs that ZMix can do:
+
+1. *Commitments* - Proof of knowledge of a secret value
+    - Required public parameters: generators, modulus/curve
+1. *Scoped Commitments* - Like *commitments* but restricted to a specific context.
+    - Required public parameters: Hash, context label, map to resulting generator, modulus
+1. *Signature Proofs of Knowledge* - Instead of revealing the signature, prove knowledge of it, allows proving knowledge of messages included in the signature. Messages can be hidden or revealed.
+    - Required public parameters: schemas, encodings, public keys, curves/modulus
+1. *Intervals* - Proof a value lies between a lower and upper bound
+    - Required public parameters:
+        - RSA: generators and modulus
+        - ECC: Bulletproofs or R1CS
+        - Circuit logic: public equations
+1. *Set Memberships* - Proof a value is in a set of known values
+    - Required public parameters:
+        - Static/Enum: Set of known values
+        - Dynamic: Accumulator with public parameters or Merkle tree with circuit logic
+1. *Verifiable Encryption* - Proof a third party knows a secret that can decrypt a value
```

Comment:
Can "correctly" decrypt a value?  Just want to avoid confusion in cases where the set of valid ciphertexts is dense in the output space (i.e. like AES).

Reply:
This would be a proof that some cipher text encrypts (part of) the witness.